### PR TITLE
Import file xls

### PR DESCRIPTION
**Pull request by Sigabort42**
Add Dictionnaire remote_access_type in file __init__.py

Add method import_xls

### CODE REVIEW

File: cbw_api_toolbox/__init__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 urllib3.disable_warnings()
 
-logging.basicConfig(level=logging.INFO)
+logging.basicConfig(level=logging.ERROR)
```
Comment (jinxka) [inline cbw_api_toolbox/__init__.py]: WHY?

File: cbw_api_toolbox/cbw_api.py [hunk 2]
```diff
@@ -167,3 +169,36 @@ def update_remote_access(self, remote_access_id, info):
...
         return False
+
+    def import_remote_accesses_xlsx(self, file_xlsx):
+        """method to import remote accesses from an xlsx file"""
+        if file_xlsx:
+            imported = xlrd.open_workbook(file_xlsx)
+            lines = imported.sheet_by_name(u'Example')
+            response = []
+
+            for line in range(1, lines.nrows):
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: type = lines.row_values(line)[3]
auth =  lines.row_values(line)[5]

File: cbw_api_toolbox/cbw_api.py [hunk 3]
```diff
@@ -167,3 +169,36 @@ def update_remote_access(self, remote_access_id, info):
...
+                info = {
+                    "address": lines.row_values(line)[0],
+                    "port": lines.row_values(line)[1],
+                    "type": REMOTE_ACCES_TYPE[
+                        lines.row_values(line)[2]][lines.row_values(line)[3]],
+                    "login": lines.row_values(line)[4],
+                    "node": lines.row_values(line)[6]
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: "password": auth if type == "PASSWORD else ""

File: cbw_api_toolbox/cbw_api.py [hunk 4]
```diff
@@ -167,3 +169,36 @@ def update_remote_access(self, remote_access_id, info):
...
+                    info["password"] = lines.row_values(line)[5]
+                    info["key"] = ""
+                else:
+                    info["password"] = ""
+                    info["key"] = lines.row_values(line)[5]
+
+                response.append(self.create_remote_access(info))
+                remote_accesses = self.remote_accesses()
+
+                for remote_access in remote_accesses:
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: Add comment

File: cbw_api_toolbox/cbw_api.py [hunk 5]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
...
+        """method to import remote accesses from an xlsx file"""
+        if file_xlsx:
+            file = file_xlsx.split(".")
+            if file[len(file) - 1] == "xlsx":
+                imported = xlrd.open_workbook(file_xlsx)
+                lines = imported.sheet_by_name(u'Example')
+                response = []
+
+                for line in range(1, lines.nrows):
+                    logging.debug("Creating remote access")
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: logging.debug("Creating remote access {}".format(address))

File: cbw_api_toolbox/cbw_api.py [hunk 6]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
...
+                        "port": lines.row_values(line)[1],
+                        "type": REMOTE_ACCES_TYPE[lines.row_values(line)[2]][type],
+                        "login": lines.row_values(line)[4],
+                        "node": lines.row_values(line)[6],
+                        "password": auth if type == "PASSWORD" else "",
+                        "key": auth if type == "KEY" else ""
+                    }
+
+                    response.append(self.create_remote_access(info))
+                    logging.debug("Created remote access")
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: "Created remote access" -> "Remote access {} successfully created".format(address)

File: cbw_api_toolbox/cbw_api.py [hunk 7]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
...
+                    remote_accesses = self.remote_accesses()
+
+                    logging.debug("Add groups for server")
+                    for remote_access in remote_accesses: #Add a group for server
+                        if remote_access.address == info["address"]:
+                            self.update_server(remote_access.server.id, lines.row_values(line)[7])
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: add log "Groups successfully added"

File: cbw_api_toolbox/cbw_api.py [hunk 8]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
 
         logging.error("Error update remote access")
         return False
+
+    def import_remote_accesses_xlsx(self, file_xlsx):
+        """method to import remote accesses from an xlsx file"""
+        if file_xlsx:
+            file = file_xlsx.split(".")
+            if file[len(file) - 1] == "xlsx":
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: add comment for this line

File: spec/test_cbw_api.py [hunk 9]
```diff
@@ -194,3 +194,23 @@ def test_update_remote_access():
             response = client.update_remote_access('4', info)
 
             assert response is False
+
+    @staticmethod
+    def test_import_remote_accesses_xlsx():
+        "Tests for metehod import_xls"
```
Comment (jinxka) [inline spec/test_cbw_api.py]: method

File: cbw_api_toolbox/cbw_api.py [hunk 10]
```diff
@@ -8,14 +8,16 @@
 from requests.exceptions import ProxyError, SSLError, RetryError, InvalidHeader, MissingSchema
 from urllib3.exceptions import NewConnectionError, MaxRetryError
 
+import xlrd
+
 from cbw_api_toolbox import API_DEFAULT_URL
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: from cbw_api_toolbox import API_DEFAULT_URL, REMOTE_ACCES_TYPE

File: cbw_api_toolbox/cbw_api.py [hunk 11]
```diff
@@ -8,14 +8,16 @@
...
 from cbw_api_toolbox.__routes__ import ROUTE_SERVERS, ROUTE_PING, ROUTE_REMOTE_ACCESSES
+from cbw_api_toolbox import REMOTE_ACCES_TYPE
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: remove this line

File: cbw_api_toolbox/cbw_api.py [hunk 12]
```diff
@@ -8,14 +8,16 @@
...
 from cbw_api_toolbox.cbw_auth import CBWAuth
 from cbw_api_toolbox.cbw_objects.cbw_server import CBWServer
 from cbw_api_toolbox.cbw_objects.cbw_remote_access import CBWRemoteAccess
 from cbw_api_toolbox.cbw_parser import CBWParser
 
-
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: why?

File: examples/import_file_xls.py [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+FILE_XLSX = ''
+
+CLIENT = CBWApi(API_URL, API_KEY, SECRET_KEY)
+
+response = CLIENT.import_remote_accesses_xlsx(FILE_XLSX)
+
+remote_accesses = CLIENT.remote_accesses()
+
+for remote_access in remote_accesses:
```
Comment (jinxka) [inline examples/import_file_xls.py]: this won't do

File: cbw_api_toolbox/cbw_api.py [hunk 14]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
...
         logging.error("Error update remote access")
         return False
+
+    def import_remote_accesses_xlsx(self, file_xlsx):
+        """method to import remote accesses from an xlsx file"""
+        if file_xlsx:
+            file = file_xlsx.split(".")
+            if file[len(file) - 1] == "xlsx":
+                imported = xlrd.open_workbook(file_xlsx)
+                lines = imported.sheet_by_name(u'Example')
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: (u'Example')?

File: cbw_api_toolbox/cbw_api.py [hunk 15]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
...
+                response = []
+
+                for line in range(1, lines.nrows):
+                    logging.debug("Creating remote access")
+                    type = lines.row_values(line)[3] # pylint: disable=redefined-builtin
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: change type to auth_type and remove pylint disable

File: cbw_api_toolbox/cbw_api.py (hunk 8 above)
Comment (fwininger) [reply cbw_api_toolbox/cbw_api.py]: Prefer : file_xlsx.endswith('.xlsx')

File: cbw_api_toolbox/cbw_api.py [hunk 16]
```diff
@@ -167,3 +169,40 @@ def update_remote_access(self, remote_access_id, info):
 
         logging.error("Error update remote access")
         return False
+
+    def import_remote_accesses_xlsx(self, file_xlsx):
+        """method to import remote accesses from an xlsx file"""
+        if file_xlsx:
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: solve your problem first : 

```
if not file_xlsx:
  logging.fatal("No Files xlsx")
  return None

if not file_xlsx.endswith('.xlsx')
  logging.fatal("Extension not valid")
  return None

...
```

File: cbw_api_toolbox/cbw_api.py (hunk 8 above)
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: endswith no exist for python3

File: cbw_api_toolbox/cbw_api.py (hunk 11 above)
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: Use REMOTE_ACCESS_TYPE

File: cbw_api_toolbox/cbw_api.py [hunk 17]
```diff
@@ -167,3 +169,42 @@ def update_remote_access(self, remote_access_id, info):
 
         logging.error("Error update remote access")
         return False
+
+    def import_remote_accesses_xlsx(self, file_xlsx):
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: I don't understand why this code is here. I don't think that the purspose of this code is "Class used to communicate with the CBW API".

You need to add a new class that read the xlsx file and use the method CbwApi.create_remote_access

File: cbw_api_toolbox/cbw_api.py [hunk 18]
```diff
@@ -167,3 +169,42 @@ def update_remote_access(self, remote_access_id, info):
...
+
+                for line in range(1, lines.nrows):
+                    address = lines.row_values(line)[0]
+                    type = lines.row_values(line)[3] # pylint: disable=redefined-builtin
+                    auth = lines.row_values(line)[5]
+                    logging.debug("Creating remote access {}".format(address))
+                    info = {
+                        "address": lines.row_values(line)[0],
+                        "port": lines.row_values(line)[1],
+                        "type": REMOTE_ACCES_TYPE[lines.row_values(line)[2]][type],
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: I prefer if you use directly the type in the excel file. The purspose of the project is just to transfert data to the API without any kink of intelligence.

File: cbw_api_toolbox/cbw_api.py [hunk 19]
```diff
@@ -167,3 +169,42 @@ def update_remote_access(self, remote_access_id, info):
...
+                        "login": lines.row_values(line)[4],
+                        "node": lines.row_values(line)[6],
+                        "password": auth if type == "PASSWORD" else "",
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: Add a column password and a column key and forward directly the data

File: cbw_api_toolbox/cbw_api.py (hunk 14 above)
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: for this file test

File: cbw_api_toolbox/cbw_api.py [hunk 20]
```diff
@@ -167,3 +169,42 @@ def update_remote_access(self, remote_access_id, info):
...
+                        "key": auth if type == "KEY" else ""
+                    }
+
+                    response.append(self.create_remote_access(info))
+                    logging.debug("Remote access {} successfully created".format(address))
+                    remote_accesses = self.remote_accesses()
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: It's a shame to browse all the remote accesses while the server id is in the create_remote_access response

File: examples/import_file_xls.py [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+
+API_KEY = ''
+SECRET_KEY = ''
+API_URL = ''
+
+FILE_XLSX = ''
+
+CLIENT = CBWApi(API_URL, API_KEY, SECRET_KEY)
+
+response = CLIENT.import_remote_accesses_xlsx(FILE_XLSX)
```
Comment (fwininger) [inline examples/import_file_xls.py]: The code for the excel part could be here.

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+                    "type": text[title.index("TYPE")] if title.index("TYPE") else "",
+                    "login": text[title.index("USERNAME")] if title.index("USERNAME") else "",
+                    "password": text[title.index("PASSWORD")] if title.index("PASSWORD") else "",
+                    "key": text[title.index("KEY")] if title.index("KEY") else "",
+                    "node": text[title.index("NODE")] if title.index("NODE") else ""
+                }
+            except ValueError:
+                logging.fatal("Error format file xlsx::"
+                              "HOST, PORT, TYPE, USERNAME, PASSWORD, KEY, NODE, GROUPS")
+                info = {
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_file_xlsx.py]: return None

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+            logging.fatal("No Files xlsx")
+            return None
+
+        file = file_xlsx.split(".")
+        if file[len(file) - 1] != "xlsx": #check if the extension is xlsx
+            logging.fatal("Extension not valid")
+            return None
+
+        imported = xlrd.open_workbook(file_xlsx)
+        lines = imported.sheet_by_name(u'Sheet1')
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_file_xlsx.py]: https://xlrd.readthedocs.io/en/latest/api.html#xlrd.book.Book.sheet_by_index

File: examples/import_file_xls.py [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+
+FILE_XLSX = ''
+
+CLIENT = CBWApi(API_URL, API_KEY, SECRET_KEY)
+
+XLSX = CBWXlsx(CLIENT)
+
+response = XLSX.import_remote_accesses_xlsx(FILE_XLSX)
+
+try:
```
Comment (jinxka) [inline examples/import_file_xls.py]: remove try and check if response != None

File: cbw_api_toolbox/cbw_file_xlsx.py (hunk 23 above)
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_file_xlsx.py]: Why ?

File: examples/import_file_xls.py [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+XLSX = CBWXlsx(API_URL, API_KEY, SECRET_KEY)
+
+response = XLSX.import_remote_accesses_xlsx(FILE_XLSX)
+
+if response:
+    for remote_access in response:
+        if remote_access:
+            print("remote access created, id=>>>>>{}".format(remote_access.id))
+        else:
+            print("Error imported")
```
Comment (jinxka) [inline examples/import_file_xls.py]: An error occurred, import_remote_accesses_xlsx failed

Comment (jinxka) [inline examples/import_file_xls.py]: An error occurred, import_remote_accesses_xlsx failed

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 26]
```diff
@@ -0,0 +1,57 @@
...
+        response = []
+        title = lines.row_values(0)
+        for line in range(1, lines.nrows):
+            text = lines.row_values(line)
+            address = lines.row_values(line)[0]
+            logging.debug("Creating remote access {}".format(address))
+
+            try:
+                info = {
+                    "address": text[title.index("HOST")] if not title.index("HOST") else "",
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_file_xlsx.py]: text[title.index("HOST")] or ""

Comment (jinxka) [inline cbw_api_toolbox/cbw_file_xlsx.py]: text[title.index("HOST")] or ""

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+
+        file = file_xlsx.split(".")
+        if file[len(file) - 1] != "xlsx": #check if the extension is xlsx
+            logging.fatal("Extension not valid")
+            return None
+
+        imported = xlrd.open_workbook(file_xlsx)
+        lines = imported.sheet_by_index(0)
+        response = []
+        title = lines.row_values(0)
```
Comment (jinxka) [inline cbw_api_toolbox/cbw_file_xlsx.py]: titles or tiles_row

File: cbw_api_toolbox/cbw_file_xlsx.py (hunk 26 above)
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_file_xlsx.py]: text[title.index("HOST")] it's better

File: spec/test_cbw_api.py [hunk 28]
```diff
@@ -194,3 +195,29 @@ def test_update_remote_access():
             response = client.update_remote_access('4', info)
 
             assert response is False
+
+    @staticmethod
+    def test_import_remote_accesses_xlsx():
+        """Tests for method import_xls"""
+        client = CBWXlsx(API_URL, API_KEY, SECRET_KEY)
+        file_xlsx = "spec/fixtures/xlsx_files/batch_import_model.xlsx"
```
Comment (fwininger) [inline spec/test_cbw_api.py]: The formatting of the files is broken.

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+        self.client = CBWApi(api_url, api_key, secret_key)
+
+    def import_remote_accesses_xlsx(self, file_xlsx):
+        """method to import remote accesses from an xlsx file"""
+        if not file_xlsx:
+            logging.fatal("No Files xlsx")
+            return None
+
+        file = file_xlsx.split(".")
+        if file[len(file) - 1] != "xlsx": #check if the extension is xlsx
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_file_xlsx.py]: again : 

```
if not file_xlsx.endswith('.xlsx')
  logging.fatal("Extension not valid")
  return None
```

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 30]
```diff
@@ -0,0 +1,58 @@
...
+            logging.fatal("Extension not valid")
+            return None
+
+        imported = xlrd.open_workbook(file_xlsx)
+        lines = imported.sheet_by_index(0)
+        response = []
+        titles = lines.row_values(0)
+        for line in range(1, lines.nrows):
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_file_xlsx.py]: add a blank line

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 31]
```diff
@@ -0,0 +1,58 @@
...
+                logging.fatal("Error format file xlsx::"
+                              "HOST, PORT, TYPE, USERNAME, PASSWORD, KEY, NODE, GROUPS")
+                return None
+
+            remote_access = self.client.create_remote_access(info)
+            response.append(remote_access)
+            logging.debug("Remote access {} successfully created".format(address))
+
+            logging.debug("Add groups for server")
+            if titles.index("GROUPS") and remote_access and remote_access.address \
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_file_xlsx.py]: Dont break the line in the middle of nowhere

File: cbw_api_toolbox/cbw_api.py [hunk 32]
```diff
@@ -124,7 +121,8 @@ def create_remote_access(self, info):
                 "node": info["node"]
             })
             logging.debug("Create connexion remote access::{}".format(response.text))
-            return self.verif_response(response)
+            if self.verif_response(response):
+                return CBWParser().parse_response(CBWRemoteAccess, response)
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: logging.info('Blabla successfully created')

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 33]
```diff
@@ -0,0 +1,58 @@
...
+                    "node": text[titles.index("NODE")]
+                }
+            except ValueError:
+                logging.fatal("Error format file xlsx::"
+                              "HOST, PORT, TYPE, USERNAME, PASSWORD, KEY, NODE, GROUPS")
+                return None
+
+            remote_access = self.client.create_remote_access(info)
+            response.append(remote_access)
+            logging.debug("Remote access {} successfully created".format(address))
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_file_xlsx.py]: remove this line, the method create_remote_access handle this message.

File: spec/test_cbw_api.py [hunk 34]
```diff
@@ -1,6 +1,7 @@
 """Test file for cbw_api.py"""
 
 from cbw_api_toolbox.cbw_api import CBWApi
+from cbw_api_toolbox.cbw_file_xlsx import CBWXlsx
```
Comment (fwininger) [inline spec/test_cbw_api.py]: why? you need to add another file for the cbw_file.xlsx tests.

File: cbw_api_toolbox/cbw_api.py [hunk 35]
```diff
@@ -3,19 +3,16 @@
 import json
 import logging
 import sys
-
```
[the remark this answers is not in the document]
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: No problem in travis :)

File: cbw_api_toolbox/cbw_api.py [hunk 36]
```diff
@@ -3,19 +3,16 @@
...
 import requests
 from requests.exceptions import ProxyError, SSLError, RetryError, InvalidHeader, MissingSchema
 from urllib3.exceptions import NewConnectionError, MaxRetryError
-
```
[the remark this answers is not in the document]
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: No problem in travis :)

File: cbw_api_toolbox/cbw_api.py [hunk 37]
```diff
@@ -3,19 +3,16 @@
...
 from cbw_api_toolbox import API_DEFAULT_URL
 from cbw_api_toolbox.__routes__ import ROUTE_SERVERS, ROUTE_PING, ROUTE_REMOTE_ACCESSES
 from cbw_api_toolbox.cbw_auth import CBWAuth
 from cbw_api_toolbox.cbw_objects.cbw_server import CBWServer
 from cbw_api_toolbox.cbw_objects.cbw_remote_access import CBWRemoteAccess
 from cbw_api_toolbox.cbw_parser import CBWParser
 
-
```
[the remark this answers is not in the document]
Comment (Sigabort42) [reply cbw_api_toolbox/cbw_api.py]: No problem in travis :)

File: cbw_api_toolbox/cbw_api.py (hunk 35 above)
[the remark this answers is not in the document]
Comment (jinxka) [reply cbw_api_toolbox/cbw_api.py]: Add new line

File: cbw_api_toolbox/cbw_api.py (hunk 36 above)
[the remark this answers is not in the document]
Comment (jinxka) [reply cbw_api_toolbox/cbw_api.py]: Add new line

File: cbw_api_toolbox/cbw_api.py (hunk 35 above)
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: add new line

File: cbw_api_toolbox/cbw_api.py (hunk 36 above)
Comment (jinxka) [inline cbw_api_toolbox/cbw_api.py]: add new line

File: cbw_api_toolbox/cbw_api.py [hunk 38]
```diff
@@ -124,7 +124,9 @@ def create_remote_access(self, info):
                 "node": info["node"]
             })
             logging.debug("Create connexion remote access::{}".format(response.text))
-            return self.verif_response(response)
+            if self.verif_response(response):
+                logging.info('remote access successfully created')
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: Ajouter l'adresse

File: cbw_api_toolbox/cbw_file_xlsx.py [hunk 39]
```diff
@@ -0,0 +1,57 @@
...
+            except ValueError:
+                logging.fatal("Error format file xlsx::"
+                              "HOST, PORT, TYPE, USERNAME, PASSWORD, KEY, NODE, GROUPS")
+                return None
+
+            remote_access = self.client.create_remote_access(info)
+            response.append(remote_access)
+            logging.debug("Add groups for server")
+
+            if titles.index("GROUPS") and remote_access \
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_file_xlsx.py]: https://www.python.org/dev/peps/pep-0008/#multiline-if-statements

Comment (Sigabort42) [reply cbw_api_toolbox/cbw_file_xlsx.py]: # Add some extra indentation on the conditional continuation line.
if (this_is_one_thing
        and that_is_another_thing):
    do_something()

if titles.index("GROUPS") and remote_access \
        and remote_access.address == info["address"]:
    self.client.update_server(remote_access.server.id, text[titles.index("GROUPS")])

Comment (fwininger) [reply cbw_api_toolbox/cbw_file_xlsx.py]: Merci de bien vouloir écrire ceci : 
```
if (titles.index("GROUPS") and remote_access
        and remote_access.address == info["address"]):
```

File: examples/import_file_xls.py [hunk 40]
```diff
@@ -0,0 +1,22 @@
+"""import file xls"""
+
+from cbw_api_toolbox.cbw_file_xlsx import CBWXlsx
+
+API_KEY = ''
+SECRET_KEY = ''
+API_URL = ''
+
+FILE_XLSX = ''
```
Comment (fwininger) [inline examples/import_file_xls.py]: ajouter un fichier `examples/import_file.xlsx` avec le bon format et mettre FILE_XLSX = 'import_file.xlsx'

File: examples/import_file_xls.py [hunk 41]
```diff
@@ -0,0 +1,22 @@
+"""import file xls"""
```
Comment (fwininger) [inline examples/import_file_xls.py]: Mettre ici xlsx et non xls. Corriger également le nom du fichier.

File: examples/import_file_xlsx.py [hunk 42]
```diff
@@ -0,0 +1,22 @@
+"""import file xlsx"""
+
+from cbw_api_toolbox.cbw_file_xlsx import CBWXlsx
+
+API_KEY = ''
+SECRET_KEY = ''
+API_URL = ''
+
+FILE_XLSX = 'examples/import_file.xlsx'
```
Comment (fwininger) [inline examples/import_file_xlsx.py]: Supprimer le fichier import_file_false.xlsx

File: spec/test_cbw_files_xlsx.py [hunk 43]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class TestCBWXlsx:
+    """Test for class CBWFilsXlsx"""
+
+    @staticmethod
+    def test_import_remote_accesses_xlsx():
+        """Tests for method import_xls"""
+        client = CBWXlsx(API_URL, API_KEY, SECRET_KEY)
+        file_xlsx = "examples/import_file.xlsx"
```
Comment (fwininger) [inline spec/test_cbw_files_xlsx.py]: Laisser les fichiers de spec dans le dossier spec.

**Pull request merged**